### ISSUE TITLE
Possible memory leak with lora usage

### ISSUE THREAD

**Issue by 0xmihutao** [comfyanonymous/ComfyUI#1462]
I use comfy as a backend in my app, and especially after using many loras, the CPU RAM usage gradually climbs. The weird part is that the RAM usage exceeds the total size of all models/loras/vaes etc.

**Comment by comfyanonymous** [comment]
What's your system? Which version of torch are you using? Which model?

**Comment by MaximeBrl** [comment]
Same problem on my GPU when I use a LORA :
'HIP out of memory. Tried to allocate 36.00 MiB (GPU 0; 11.98 GiB total capacity; 11.51 GiB already allocated; 178.00 MiB free; 11.75 GiB reserved in total by PyTorch) If reserved memory is >> allocated memory try setting max_split_size_mb to avoid fragmentation.  See documentation for Memory Management and PYTORCH_HIP_ALLOC_CONF'

I am on LINUX. AMD 6750XT & 16GB RAM

**Comment by 0xmihutao** [comment]
@comfyanonymous I'm running on ubuntu 20.04, 12 vcpu, 31 gb ram (cpu), A4500 (20GB VRAM), pytorch 2.0.1. I'm using a variety of loras from civitai on top of cyberrealistic base model

**Comment by comfyanonymous** [comment]
Are you using them at the same time? Do you use the default "Load Lora" node or a custom node?

**Comment by 0xmihutao** [comment]
Yes I'm using multiple at the same time, with the default load lora node (chained in a row).

**Comment by le-khang** [comment]
That's exactly the issue I'm facing right now. I have around 30 Loras in my workflows and the CPU RAM usage gradually climbs until ComfyUI crashes. Is there any way to fix this please?

**Comment by ashleykleynhans** [comment]
A1111 also faced a similar issue with switching between Stable Diffusion checkpoints on Linux and on Ubuntu, it was solved as follows:

> [!NOTE]  
> Drop the `sudo` if you are running on an environment like RunPod where you are already root.

```
sudo apt update
sudo apt install libtcmalloc-minimal4
TCMALLOC="$(ldconfig -p | grep -Po "libtcmalloc.so.\d" | head -n 1)"
export LD_PRELOAD="${TCMALLOC}"
```

And then start ComfyUI as normal.

I am not sure whether it will resolve the issue for ComfyUI, but it worked for A1111.

**Comment by le-khang** [comment]
Yeah and here is Ashleyk's code for his template on Runpod if anybody needs it. It's working great for me.

fuser -k 3021/tcp
TCMALLOC="$(ldconfig -p | grep -Po "libtcmalloc.so.\d" | head -n 1)"
export LD_PRELOAD="${TCMALLOC}"
cd /workspace/ComfyUI
source venv/bin/activate
python3 main.py --listen 0.0.0.0 --port 3021 > /workspace/logs/comfyui.log 2>&1 &

Thank you Ashleyk.

**Comment by ashleykleynhans** [comment]
> Yeah and here is Ashleyk's code for his template on Runpod if anybody needs it. It's working great for me.
> 
> fuser -k 3021/tcp TCMALLOC="$(ldconfig -p | grep -Po "libtcmalloc.so.\d" | head -n 1)" export LD_PRELOAD="${TCMALLOC}" cd /workspace/ComfyUI source venv/bin/activate python3 main.py --listen 0.0.0.0 --port 3021 > /workspace/logs/comfyui.log 2>&1 &
> 
> Thank you Ashleyk.

Note that this will only work with my specific template on RunPod ("Stable Diffusion Kohya_ss ComfyUI Ultimate"), it will not work with any of the other ComfyUI templates on RunPod because they run on different ports.

I've also updated my template (version 3.3.2) to load libtcmalloc before starting ComfyUI, so these steps should not be necessary for anybody deploying a new pod on RunPod.
